# **THE SPARKS FOUNDATION**
# **Author: Manasi Chhibber**
# **Data Science & Business Analytics Intern**

## **Task #1: Prediction using Supervised ML**
### Predicting the score of a student based on the no. of study hours using Linear Regression.
## **Problem Statement: What will be predicted score if a student studies for 9.25 hrs/day?**

### Importing Libraries

In [7]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

### Getting Data

In [2]:
df = pd.read_csv('/content/task1.csv')
df

Hours  Scores
0     2.5      21
1     5.1      47
2     3.2      27
3     8.5      75
4     3.5      30
5     1.5      20
6     9.2      88
7     5.5      60
8     8.3      81
9     2.7      25
10    7.7      85
11    5.9      62
12    4.5      41
13    3.3      42
14    1.1      17
15    8.9      95
16    2.5      30
17    1.9      24
18    6.1      67
19    7.4      69
20    2.7      30
21    4.8      54
22    3.8      35
23    6.9      76
24    7.8      86

### Splitting Data for Training and Testing

In [4]:
data = df.values
X, y = data[:,:-1], data[:,-1] 
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.20, random_state=10)  # splitting in the ratio 80:20

### Training

In [5]:
model = LinearRegression()
model.fit(X_train, y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

### Making Predictions

In [6]:
y_pred = model.predict(X_test)

### Accuracy Check

In [8]:
mean_squared_error(y_test, y_pred)  # using mean squared error for checking accuracy

34.30075055068739

### Solution to Problem Statement

In [9]:
model.predict([[9.25]])

array([93.43229054])

## **Result: A student studying for 9.25 hrs/day should score 93.43.**